src/System.Buffers.Experimental/System/Buffers/Native/NativeMemoryPool.cs

```diff
@@ -24,7 +24,7 @@ public unsafe sealed partial class NativeMemoryPool : MemoryPool<byte>
         int _bufferCount;
         bool[] _rented;
 
-        public static NativeMemoryPool Shared => s_shared;
+        public static new NativeMemoryPool Shared => s_shared;
```

Comment:
I don't understand the reasoning for this. Why is the new keyword required here (i.e. why do we get a warning otherwise)?

Reply:
The base class `MemoryPool` has a property with the same name:
```c#
        /// <summary>
        /// Returns a singleton instance of a MemoryPool based on arrays.
        /// </summary>
        public static MemoryPool<T> Shared => s_shared;
```
This one hides it. You are allowed to do that but it will warn unless you indicate using the `new` keyword that you intended to do it. I am not sure why it is not an error.

The advantage of `new` over `virtual`/`overrides` is presumably that there is no virtual dispatch overhead.